pkg/provider/alibaba/client.go

```diff
@@ -0,0 +1,253 @@
+	return s.endpoint
+}
+
+func (s *secretsManagerClient) GetSecretValue(
+	ctx context.Context,
+	request *kms.GetSecretValueRequest,
+) (*kms.GetSecretValueResponseBody, error) {
+	resp, err := s.doApiCall(ctx, "GetSecretValue", request)
+	if err != nil {
+		return nil, fmt.Errorf("error getting secret [%s] latest value: %w", utils.Deref(request.SecretName), err)
+	}
+
+	body, err := utils.ConvertToType[kms.GetSecretValueResponseBody](resp)
+	if err != nil {
+		return nil, fmt.Errorf("error converting body: %w", err)
+	}
+
+	return &body, nil
+}
+
+func (s *secretsManagerClient) doApiCall(ctx context.Context,
+	action string,
+	request interface{}) (interface{}, error) {
+
+	accessKeyID, err := s.config.Credential.GetAccessKeyId()
+	if err != nil {
+		return nil, fmt.Errorf("error getting AccessKeyId: %w", err)
+	}
+
+	accessKeySecret, err := s.config.Credential.GetAccessKeySecret()
+	if err != nil {
+		return nil, fmt.Errorf("error getting AccessKeySecret: %w", err)
+	}
+
+	securityToken, err := s.config.Credential.GetSecurityToken()
+	if err != nil {
+		return nil, fmt.Errorf("error getting SecurityToken: %w", err)
+	}
+
+	apiRequest := newOpenAPIRequest(s.endpoint, action, methodTypeGET, request)
+	apiRequest.query["AccessKeyId"] = accessKeyID
+
+	if utils.Deref(securityToken) != "" {
+		apiRequest.query["SecurityToken"] = securityToken
+	}
+
+	apiRequest.query["Signature"] = openapiutil.GetRPCSignature(apiRequest.query, utils.Ptr(apiRequest.method.String()), accessKeySecret)
+
+	httpReq, err := newHttpRequestWithContext(ctx, apiRequest)
+	if err != nil {
+		return nil, fmt.Errorf("error creating http request: %w", err)
+	}
+
+	resp, err := s.client.Do(httpReq)
+	if err != nil {
+		return nil, fmt.Errorf("error invoking http request: %w", err)
+	}
+
+	return s.parseResponse(resp)
+}
+
+func (s *secretsManagerClient) parseResponse(resp *http.Response) (map[string]interface{}, error) {
+	statusCode := utils.Ptr(resp.StatusCode)
+	if utils.Deref(util.Is4xx(statusCode)) || utils.Deref(util.Is5xx(statusCode)) {
+		return nil, s.parseErrorResponse(resp)
+	}
+
+	obj, err := util.ReadAsJSON(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := util.AssertAsMap(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+func (s *secretsManagerClient) parseErrorResponse(resp *http.Response) error {
+	res, err := util.ReadAsJSON(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	errorMap, err := util.AssertAsMap(res)
+	if err != nil {
+		return err
+	}
+
+	requestId := defaultAny(errorMap["RequestId"], errorMap["requestId"])
+	errorMap["statusCode"] = utils.Ptr(resp.StatusCode)
+	err = tea.NewSDKError(map[string]interface{}{
+		"code":               tea.ToString(defaultAny(errorMap["Code"], errorMap["code"])),
+		"message":            fmt.Sprintf("code: %s, %s request id: %s", tea.ToString(resp.StatusCode), tea.ToString(defaultAny(errorMap["Message"], errorMap["message"])), tea.ToString(requestId)),
```

Comment:
please remove the `requestID` from the logs, this message bubbles up and is emitted as an event. Duplicate events are aggregated, but in this case they can not due to the request-id in the message: 

![event-flood](https://user-images.githubusercontent.com/1709030/212765534-ae3f5220-9d24-44c8-9d5c-28e3a88aa867.png)

Reply:
removed in the errros I created, we stilll have it for auth errors of the SDK